Correction about generation promotions

amazur31 opened the pull request:
You can't promote to generation 2 from generation 2.

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

amazur31 committed:
Correction about generation promotions

You can't promote to generation 2 from generation 2.

gfoidl commented on docs/standard/garbage-collection/large-object-heap.md:
```diff
@@ -32,7 +32,7 @@ Generations provide a logical view of the GC heap. Physically, objects live in m
 For the SOH, objects that survive a GC are promoted to the next generation. Objects that survive a generation 0 collection are now considered generation 1 objects, and so on. However, objects that survive the oldest generation are still considered to be in the oldest generation. In other words, survivors from generation 2 are generation 2 objects; and survivors from the LOH are LOH objects (which ...
 
-User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 1 and 2).
+User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 0 and 1).
```
```suggestion
User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 1).
```
? An object can only be in Gen-1 if promoted from Gen-0 previously.

BTW: This paragraph is missing the POH (.NET 5+).

amazur31 committed:
Update docs/standard/garbage-collection/large-object-heap.md

"generations 1" <- 0 and 1, since generation 2 cleans both generation 0 and then generation 1, so survivors from both 0 and 1

Co-authored-by: Günther Foidl <gue@korporal.at>

gewarren commented on docs/standard/garbage-collection/large-object-heap.md:
```diff
@@ -32,7 +32,7 @@ Generations provide a logical view of the GC heap. Physically, objects live in m
 For the SOH, objects that survive a GC are promoted to the next generation. Objects that survive a generation 0 collection are now considered generation 1 objects, and so on. However, objects that survive the oldest generation are still considered to be in the oldest generation. In other words, survivors from generation 2 are generation 2 objects; and survivors from the LOH are LOH objects (which ...
 
-User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 1 and 2).
+User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 1).
```
```suggestion
User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generation 1).
```

amazur31 committed:
typo fix

gewarren commented:
Thanks @amazur31 !
